gqcp/include/Basis/SpinorBasis/RSpinOrbitalBasis.hpp

```diff
@@ -225,25 +232,30 @@ class RSpinOrbitalBasis:
      */
 
     /**
-     *  @param ao_list     indices of the AOs used for the Mulliken populations
-     *
-     *  @return the Mulliken operator for a set of given AO indices
-     *
-     *  @note this method is only available for real matrix representations
+     *  Partition this set of restricted spin-orbitals according to the Mulliken partitioning scheme.
+     * 
+     *  @param selector             A function that returns true for basis functions that should be included the Mulliken partitioning.
+     * 
+     *  @return A `RMullikenPartitioning` for the AOs selected by the supplied selector function.
      */
-    template <typename S = ExpansionScalar, typename = IsReal<S>>
-    ScalarRSQOneElectronOperator<double> calculateMullikenOperator(const std::vector<size_t>& ao_list) const {
+    RMullikenPartitioning<ExpansionScalar> mullikenPartitioning(const std::function<bool(const BasisFunction&)>& selector) const {
```

Comment:
Does the selector mean that we no longer have to specify the "Mulliken indices" in our calculations? That is an awesome improvement!

Reply:
Yes, you can check an example in the tests! :)